### PR TITLE
Lavezzo2020suppression

### PR DESCRIPTION
**Pull request by haisuzhang**
test

### CODE REVIEW

**Comment by tillahoffmann** [comment]
Thank you for opening the PR. Could you rename `data/lavezzo2020suppression/lavezzo2020suppression.md` to `data/lavezzo2020suppression/lavezzo2020suppression-extraction.md`? Then we'll be able to make sure the extraction script is reproducible.

File: data/lavezzo2020suppression/lavezzo2020suppression-extraction.md
Comment (tillahoffmann) [inline data/lavezzo2020suppression/lavezzo2020suppression-extraction.md:206]:
```suggestion
haisuproj = dict(title="Suppression of a SARS-CoV-2 outbreak in the Italian municipality of Vo",
```

Maybe we can drop this last character here? It ends up being strangely in the yaml file.

File: data/lavezzo2020suppression/lavezzo2020suppression-extraction.md
Comment (tillahoffmann) [inline data/lavezzo2020suppression/lavezzo2020suppression-extraction.md:249]:
```suggestion
```

The `with` statement automatically closes the file, so we can drop this line.

File: data/lavezzo2020suppression/lavezzo2020suppression-extraction.md
Comment (tillahoffmann) [inline data/lavezzo2020suppression/lavezzo2020suppression-extraction.md:206]: Let's rename `haisuproj` to a more descriptive variable name.

File: data/lavezzo2020suppression/lavezzo2020suppression-extraction.md
Comment (tillahoffmann) [inline data/lavezzo2020suppression/lavezzo2020suppression-extraction.md:33]:
```suggestion
from shedding_hub import folded_str
```

We can use the common implementation from the package now.

File: data/lavezzo2020suppression/lavezzo2020suppression-extraction.md
Comment (tillahoffmann) [inline data/lavezzo2020suppression/lavezzo2020suppression-extraction.md:38]: Maybe we can name this `df_demographics`? It took me a little bit to figure out what the "demo" meant.

File: data/lavezzo2020suppression/lavezzo2020suppression-extraction.md
Comment (tillahoffmann) [inline data/lavezzo2020suppression/lavezzo2020suppression-extraction.md:47]:
```suggestion
df_demo2.describe()
```

File: data/lavezzo2020suppression/lavezzo2020suppression-extraction.md
Comment (tillahoffmann) [inline data/lavezzo2020suppression/lavezzo2020suppression-extraction.md:192]: What are your thoughts on reporting `age_group` instead of `age` here? People might expect the age to be the exact age in years rather than the beginning of an age group. This could be particularly important for pediatric patients.

File: data/lavezzo2020suppression/lavezzo2020suppression-extraction.md
Comment (tillahoffmann) [inline data/lavezzo2020suppression/lavezzo2020suppression-extraction.md:46]: It looks like both `df_demo` and `df_demo2` are used in the code below. Would it make sense to just replace `df_demo` with the filtered value or do we need both the filtered and unfiltered data later?

**Review by tillahoffmann — COMMENTED** [review]
This looks great, I've left a few inline comments.

Once we're ready to merge the PR, let's "squash-merge" which eliminates files that were added and removed in different commits. That option can be selected in the drop-down of the merge button.

image

File: data/lavezzo2020suppression/lavezzo2020suppression-extraction.md
Comment (YWAN446) [inline data/lavezzo2020suppression/lavezzo2020suppression-extraction.md:19]: Are all numpy, textwrap, and datetime used?

File: data/lavezzo2020suppression/lavezzo2020suppression-extraction.md
Comment (YWAN446) [reply data/lavezzo2020suppression/lavezzo2020suppression-extraction.md:26]: Agree. It would be better to give more informative names to those dataframes.

File: data/lavezzo2020suppression/lavezzo2020suppression-extraction.md
Comment (YWAN446) [reply data/lavezzo2020suppression/lavezzo2020suppression-extraction.md:180]: That makes sense. Sorry, I think I told Haisu to use the average. @tillahoffmann Do we need to modify the schema for this?

File: data/lavezzo2020suppression/lavezzo2020suppression-extraction.md
Comment (YWAN446) [inline data/lavezzo2020suppression/lavezzo2020suppression-extraction.md:236]: I rename the dataframe from "haisuproj" to "Lavezzo2020".

File: data/lavezzo2020suppression/lavezzo2020suppression.yaml
Comment (YWAN446) [inline data/lavezzo2020suppression/lavezzo2020suppression.yaml:71]: The data were double-entered. The numbers in this one are the same as the last person.

**Review by YWAN446 — COMMENTED** [review]
Great progress, Haisu! 
Please go through the comments from Till and me. Your study is much more challenging compared to others' assignments. I did some spot checking and the values and time look good to me (except many records were double-entered). The YAML file is pretty close. 
After that, you may need to spend some time to clean up the codes (by adding more comments, removing those codes are not needed, and adding some markdown text).

File: data/lavezzo2020suppression/lavezzo2020suppression-extraction.md
Comment (tillahoffmann) [reply data/lavezzo2020suppression/lavezzo2020suppression-extraction.md:180]: If we include the attribute as `age_group`, we won't need to modify the schema. If we want to add `0-17` to the `age` field, we'll have to make a small update. What do you think is better?

Comment (YWAN446) [reply data/lavezzo2020suppression/lavezzo2020suppression-extraction.md:180]: I would recommend creating `age_group` and keeping `age` as numeric, which could be less confusing for users with less programming background.

Comment (haisuzhang) [reply data/lavezzo2020suppression/lavezzo2020suppression-extraction.md]: Age group were kept as original value, for example, 61-70.

File: data/lavezzo2020suppression/lavezzo2020suppression.yaml
Comment (haisuzhang) [reply data/lavezzo2020suppression/lavezzo2020suppression.yaml:71]: This bug has been addressed.

File: data/lavezzo2020suppression/lavezzo2020suppression-extraction.md
Comment (haisuzhang) [reply data/lavezzo2020suppression/lavezzo2020suppression-extraction.md]: Fixed

File: data/lavezzo2020suppression/lavezzo2020suppression-extraction.md
Comment (haisuzhang) [reply data/lavezzo2020suppression/lavezzo2020suppression-extraction.md]: df_demo was used in later codes so that this naming is not changed.

File: data/lavezzo2020suppression/lavezzo2020suppression-extraction.md
Comment (haisuzhang) [reply data/lavezzo2020suppression/lavezzo2020suppression-extraction.md:24]: Yes

File: data/lavezzo2020suppression/lavezzo2020suppression-extraction.md
Comment (haisuzhang) [reply data/lavezzo2020suppression/lavezzo2020suppression-extraction.md]: Renamed

**Review by YWAN446 — APPROVED** [review]
Great job!

**Pull request merged**
